pkg/kubectl/apply/element.go

```diff
@@ -325,6 +326,17 @@ func IsAdd(e Element) bool {
 	return e.HasLocal() && !e.HasRemote()
 }
 
+// IsConflict returns true if the field represented by e has conflict between remote and recorded value
+func IsConflict(e Element) bool {
+	if !e.HasRecorded() && !e.HasRemote() {
+		return false
+	}
+	if e.HasRecorded() && e.HasRemote() {
+		return !reflect.DeepEqual(e.GetRecorded(), e.GetRemote())
```

Comment:
We should only do this check on PrimitiveElements, it doesn't make sense for maps or lists